Yoast/wpseo-news
# CS/QA: minor regex improvements

## Summary
This PR can be summarized in the following changelog entry:
* Fixed an issue where image information retrieved and used in the sitemap could be incomplete.

## Relevant technical choices:

The original regex could match mis-matched quotes. The improved regexes won't.

Example:
Given the following string `alt="something he's said"`, the original regex would match `alt="something he'`, the improved regex will match `alt="something he's said"` correctly.

Given the following string `alt="something he\"s said"`, the original regex would match `alt="something he\"`, the improved regex will match `alt="something he\"s said"` correctly.


## Test instructions
* Start with the `trunk` branch
* Add a post with an image with a single/double qoute in the alt/title tag.
* Open the news image sitemap and take note of the incorrect information in the `<image:title>` and `<image:caption>` tags.
* Switch to this branch.
* Clear the sitemap cache.
* Open the new image sitemap and see the difference.
